logviewer/src/main/java/org/togetherjava/logwatcher/views/usermanagement/UserManagement.java

```diff
@@ -0,0 +1,200 @@
+package org.togetherjava.logwatcher.views.usermanagement;
+
+import com.vaadin.componentfactory.EnhancedDialog;
+import com.vaadin.flow.component.ClickEvent;
+import com.vaadin.flow.component.Text;
+import com.vaadin.flow.component.button.Button;
+import com.vaadin.flow.component.checkbox.CheckboxGroup;
+import com.vaadin.flow.component.grid.Grid;
+import com.vaadin.flow.component.grid.contextmenu.GridContextMenu;
+import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
+import com.vaadin.flow.component.orderedlayout.VerticalLayout;
+import com.vaadin.flow.component.textfield.NumberField;
+import com.vaadin.flow.component.textfield.TextField;
+import com.vaadin.flow.data.provider.DataProvider;
+import com.vaadin.flow.data.provider.Query;
+import com.vaadin.flow.router.PageTitle;
+import com.vaadin.flow.router.Route;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.logwatcher.accesscontrol.AllowedRoles;
+import org.togetherjava.logwatcher.accesscontrol.Role;
+import org.togetherjava.logwatcher.entities.UserDTO;
+import org.togetherjava.logwatcher.users.IUserRepository;
+import org.togetherjava.logwatcher.util.NotificationUtils;
+import org.togetherjava.logwatcher.views.MainLayout;
+import org.togetherjava.tjbot.db.DatabaseException;
+import org.togetherjava.tjbot.db.generated.tables.pojos.Users;
+
+import javax.annotation.security.PermitAll;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+/**
+ * The User-Management View in the Browser
+ */
+
+@PageTitle("UserManagement")
+@Route(value = "user", layout = MainLayout.class)
+@AllowedRoles(roles = {Role.ADMIN})
+@PermitAll
+public class UserManagement extends VerticalLayout {
+
+    private final transient IUserRepository repo;
+    private final Grid<UserDTO> grid = new Grid<>(UserDTO.class, false);
+
+    public UserManagement(IUserRepository repository) {
+        this.repo = repository;
+        add(grid);
+
+
+        GridContextMenu<UserDTO> menu = grid.addContextMenu();
+        menu.addItem("Add", this::onAdd);
+        menu.addItem("Edit", this::onEdit);
+        menu.addItem("Delete", this::onDelete);
+
+        grid.setColumns("discordID", "userName", "roles");
+        grid.setItems(DataProvider.fromCallbacks(this::onAll, query -> (int) onAll(query).count()));
```

Comment:
Is this `int` case safe? What if its too big? At least please add a comment explaining why this cant be the case. Otherwise the next reader will be confused about this.

Reply:
DataProvider expect's an Integer, and I don't think were are gonna display more an 2 Billion Log-Entries at once, but I add an comment

Reply:
Makes sense. A comment saying exactly what you just said will be helpful in that case, imo